src/main/java/pl/wrapper/parking/facade/domain/ParkingServiceImpl.java

```diff
@@ -1,16 +1,95 @@
 package pl.wrapper.parking.facade.domain;
 
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 import pl.wrapper.parking.facade.ParkingService;
+import pl.wrapper.parking.infrastructure.error.ParkingError;
+import pl.wrapper.parking.infrastructure.error.Result;
 import pl.wrapper.parking.pwrResponseHandler.PwrApiServerCaller;
+import pl.wrapper.parking.pwrResponseHandler.dto.ParkingResponse;
 
 @Service
+@Slf4j
 record ParkingServiceImpl(PwrApiServerCaller pwrApiServerCaller) implements ParkingService {
-    // 1. endpointy:
-    // - endpoint dla każdego parametru, zwraca jeden, kilka, lub nic
-    // - research, closest address
-    // - most free spots, everything that has => free spots, eveything has any free spots
-    // -* error handling, try to use as few exceptions as possible (think of a good way to do it)
-    //   dokumentacja + swagger do enpointa/endpointów i testy jednostkowe + test integracyjny dla waszego
-    // enpointa/enpoiontów
+
+    @Override
+    public Result<ParkingResponse> getByName(String name, Boolean opened) {
+        Optional<ParkingResponse> found = findParking(parking -> Objects.equals(name, parking.name()),opened);
+        log.info("All parkings has been checked for name");
+
+        if (found.isPresent()) return handleFoundParking(found.get());
```

Comment:
We should use the in-built Optional functionality instead of isPresent checks: 
return /methodToFindParkingThatReturnsOptional/.map(Result::success).orElse(Result.failure(/someError/));